service/frontend/adminHandler.go

```diff
@@ -245,23 +242,25 @@ func (adh *adminHandlerImpl) AddSearchAttribute(
 		adh.GetLogger().Warn("Failed to update dynamicconfig. This is only useful in local dev environment for filebased config. Please ignore this warn if this is in a real Cluster, because your filebased dynamicconfig MUST be updated separately. Configstore dynamic config will also require separate updating via the CLI.")
 	}
 
-	// update elasticsearch mapping, new added field will not be able to remove or update
-	index := adh.params.ESConfig.GetVisibilityIndex()
-	for k, v := range searchAttr {
-		valueType := convertIndexedValueTypeToESDataType(v)
-		if len(valueType) == 0 {
-			return adh.error(&types.BadRequestError{Message: fmt.Sprintf("Unknown value type, %v", v)}, scope)
-		}
-		err := adh.params.ESClient.PutMapping(ctx, index, definition.Attr, k, valueType)
-		if adh.esClient.IsNotFoundError(err) {
-			err = adh.params.ESClient.CreateIndex(ctx, index)
+	// when have valid advance visibility config, update elasticsearch mapping, new added field will not be able to remove or update
+	if err := adh.validateConfigForAdvanceVisibility(); err == nil {
```

Comment:
The only downside of this approach is that NOOP will never be logged/reported, maybe add warn level message when validation is failing?

Reply:
Added warning "Skip updating OpenSearch/ElasticSearch mapping since Advance Visibility hasn't been enabled." when Advance Visibility is not enabled.